pkg/management/postgres/metrics/collector.go

```diff
@@ -378,6 +387,65 @@ func (c QueryCollector) collect(conn *sql.DB, ch chan<- prometheus.Metric) error
 	return nil
 }
 
+// isCollectable checks if a query to collect metrics can be executed or not.
+// The method tests the query provided in the PredicateQuery property within the same transaction
+// used to collect metrics.
+// Accepts a query that returns a boolean on the first column of the first row.
+func (c QueryCollector) isCollectable(tx *sql.Tx) (bool, error) {
+	if c.userQuery.PredicateQuery == "" {
+		return true, nil
+	}
+
+	rows, err := tx.Query(c.userQuery.PredicateQuery)
+	if err != nil {
+		return false, err
+	}
+
+	defer func() {
+		if err = rows.Close(); err != nil {
+			log.Warning("Error while closing predicate evaluation",
+				"err", err.Error())
+		}
+	}()
+
+	columns, err := rows.Columns()
+	if err != nil {
+		return false, err
+	}
+
+	columnData := make([]interface{}, len(columns))
+	scanArgs := make([]interface{}, len(columns))
+	for i := range columnData {
+		scanArgs[i] = &columnData[i]
+	}
+
+	if !rows.Next() {
+		return false, nil
+	}
+	if err = rows.Scan(scanArgs...); err != nil {
+		return false, err
+	}
+	if err = rows.Err(); err != nil {
+		log.Warning("Error while loading predicate evaluation",
+			"err", err.Error())
+		return false, nil
+	}
+
+	if len(columnData) == 0 || columnData[0] == nil {
+		return false, nil
+	}
+
+	result, ok := columnData[0].(bool)
+	if !ok {
+		log.Warning("Unsupported column type",
+			"predicate_query", c.userQuery.PredicateQuery,
+			"type", fmt.Sprintf("%T", result))
+		return false, nil
+	}
```

Comment:
why not use `QueryRow` and scan into a boolean, and then the driver will let us know the first argument on the firs row is not a predicate?

Reply:
I don't think it's too much to as, given a field called `predicate_query`, that the query should be a predicate.